runtime/common/src/migrations.rs

```diff
@@ -59,6 +64,146 @@ where
 	}
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct OldRoundInfo<BlockNumber> {
+	pub current: RoundIndex,
+	pub first: BlockNumber,
+	pub length: u32,
+}
+pub struct UpdateFirstRoundNumberValue<T>(pub PhantomData<T>);
+impl<T> Migration for UpdateFirstRoundNumberValue<T>
+where
+	T: pallet_parachain_staking::Config,
+	T: pallet_async_backing::Config,
+	T: frame_system::Config,
+	u32: From<<<<T as frame_system::Config>::Block as BlockT>::Header as HeaderT>::Number>,
+{
+	fn friendly_name(&self) -> &str {
+		"MM_UpdateFirstRoundNumberValue"
+	}
+
+	fn migrate(&self, _available_weight: Weight) -> Weight {
+		let _ = Round::<T>::translate::<OldRoundInfo<BlockNumberFor<T>>, _>(|v0| {
+			let old_current = v0
+				.expect("old current round value must be present!")
+				.current;
+			let old_first: u32 = v0.expect("old first should be present!").first.into();
+			let old_length = v0.expect("old round length value must be present!").length;
+
+			// Fetch the last parachain block
+			let para_block: u32 = frame_system::Pallet::<T>::block_number().into();
+
+			// Calculate how many blocks have passed so far in this round
+			let para_block_diff: u64 = para_block.saturating_sub(old_first).into();
+
+			// Read the last relay slot from the SlotInfo storage
+			let relay_slot = pallet_async_backing::Pallet::<T>::slot_info()
+				.unwrap_or((Slot::from(284_000_000u64), 0u32))
+				.0;
+
+			// Calculate the new first
+			let new_first = u64::from(relay_slot).saturating_sub(para_block_diff);
+
+			Some(RoundInfo {
```

Comment:
I understood this a bit differently from what is done here, so have some questions:
1. Why is not current the relay slot now?
2. Shouldn't the length be the new round length?
3. Shouldn't the first be computed taking into account the new length vs the old length and use the % of the slot completed?

Reply:
1 - Because `current` is the number of the current round, not the first block or slot where it began. We want to use the slot to compute the new `first` of the round.
2 - The length is the same old one because it doesn't have to change. Basically, this migration was done assuming that we will enable async backing on Moonbase. So, when we enable it, we will have a difference of two between each relay slot received, and with an average of 6 seconds per block, we end up having the same round length (in seconds).
3 - As the the new length and the old length are the same (in seconds) we don't need to calculate any percentage, we just need to calculate how many para-blocks have passed so far in this round and substract that diff from the last relay slot fetched.